auth-starter/src/main/java/com/vaadin/auth/starter/VaadinAuthSecurityConfiguration.java

```diff
@@ -0,0 +1,66 @@
+package com.vaadin.auth.starter;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.oauth2.client.OAuth2LoginConfigurer;
+import org.springframework.util.StringUtils;
+
+import com.vaadin.flow.server.auth.ViewAccessChecker;
+import com.vaadin.flow.spring.security.VaadinWebSecurity;
+
+/**
+ * This configuration bean is provided to auto-configure Vaadin and Spring to
+ * allow single sign-on against external identity providers.
+ * <p>
+ * It enables OAuth2/OpenID login for the providers defined in the current
+ * application configuration and enabled the {@link ViewAccessChecker} for the
+ * configured login route (which defaults to {@code /login}.
```

Comment:
I think this should be `enables` as well, but maybe it can be omitted here. 
```suggestion
 * It enables OAuth2/OpenID login for the providers defined in the current
 * application configuration and the {@link ViewAccessChecker} for the
 * configured login route (which defaults to {@code /login}.
```

Reply:
Yeah, and it's also a bit confusing. The `ViewAccessChecker` is an implementation details, we don't event need to mention it in the JavaDoc: I rephrased to make it more clear!